process/notifier/headersNotifier.go

```diff
@@ -0,0 +1,62 @@
+package notifier
+
+import (
+	"encoding/hex"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-sovereign-notifier-go/process"
+)
+
+type headersNotifier struct {
+	mutSubscribers sync.RWMutex
+	subscribers    []process.IncomingHeaderSubscriber
+	marshaller     marshal.Marshalizer
+	hasher         hashing.Hasher
+}
+
+func newHeadersNotifier(marshaller marshal.Marshalizer, hasher hashing.Hasher) *headersNotifier {
+	return &headersNotifier{
+		mutSubscribers: sync.RWMutex{},
+		subscribers:    make([]process.IncomingHeaderSubscriber, 0),
+		marshaller:     marshaller,
+		hasher:         hasher,
+	}
+}
+
+func (hn *headersNotifier) registerSubscriber(handler process.IncomingHeaderSubscriber) error {
+	if check.IfNil(handler) {
+		return errNilHeaderSubscriber
+	}
+
+	hn.mutSubscribers.Lock()
+	hn.subscribers = append(hn.subscribers, handler)
+	hn.mutSubscribers.Unlock()
+
+	return nil
+}
+
+func (hn *headersNotifier) notifyHeaderSubscribers(header sovereign.IncomingHeaderHandler) error {
+	headerHash, err := core.CalculateHash(hn.marshaller, hn.hasher, header)
```

Comment:
could hasher and marshalizer be removed from here and the notifyHeaderSubscribers(header sovereign.IncomingHeaderHandler) contain also the headerhash?

Reply:
ok